train_search_multi_dataset.sh

```
@@ -0,0 +1,293 @@
+  local val_instances_per_dataset=512
+  local total_val_groups=$((${#TEST_DATASETS[@]} * val_instances_per_dataset))
+  
+  # Convert test datasets to tags with 512 groups per dataset
+  # Use dataset names directly instead of capitalized tags
+  local test_tags=()
+  local test_n_groups=()
+  for test_ds in "${TEST_DATASETS[@]}"; do
+    test_tags+=("$test_ds")  # Use dataset name directly
+    test_n_groups+=("${val_instances_per_dataset}")
+  done
+  local test_tags_str=$(IFS=,; echo "${test_tags[*]}")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)